### ISSUE TITLE
Cookies.config references global state.

### ISSUE THREAD

**Issue by Jacoby6000** [playframework/playframework#9214]
### Play Version

2.7.1

### API

Scala

### Operating System (Ubuntu 15.10 / MacOS 10.10 / Windows 10)

All

### JDK (Oracle 1.8.0_72, OpenJDK 1.8.x, Azul Zing)

All

### Expected Behavior

methods like `mergeCookieHeader` should not reference global configuration.  Global configuration is deprecated.

### Actual Behavior

If you call a method on the `Cookies` companion object which eventually references the `config` defined in `Cookies`, you will get an error regarding the use of the Global state.

```
java.lang.RuntimeException: The global application reference is disabled. Play's global state is deprecated and will
[info] be removed in a future release. You should use dependency injection instead. To enable
[info] the global application anyway, set play.allowGlobalApplication = true.
[info]   at scala.sys.package$.error(package.scala:30)
[info]   at play.api.Play$.privateMaybeApplication(Play.scala:90)
[info]   at play.api.http.HttpConfiguration$.current(HttpConfiguration.scala:361)
[info]   at play.api.mvc.Cookies$.config(Cookie.scala:159)
[info]   at play.api.mvc.CookieHeaderEncoding.encodeCookieHeader(Cookie.scala:261)
[info]   at play.api.mvc.CookieHeaderEncoding.encodeCookieHeader$(Cookie.scala:260)
[info]   at play.api.mvc.Cookies$.encodeCookieHeader(Cookie.scala:155)
[info]   at play.api.mvc.CookieHeaderEncoding.mergeCookieHeader(Cookie.scala:349)
[info]   at play.api.mvc.CookieHeaderEncoding.mergeCookieHeader$(Cookie.scala:346)
[info]   at play.api.mvc.Cookies$.mergeCookieHeader(Cookie.scala:155)
```

### Reproducible Test Case

Just call any method which references `Cookies.config`. `mergeCookieHeader` for example.

Here's the cause: https://github.com/playframework/playframework/blob/2.7.1/core/play/src/main/scala/play/api/mvc/Cookie.scala#L159

**Comment by marcospereira** [comment]
Hey @Jacoby6000, the problem here is a missing `@deprecated` annotation on `play.api.mvc.Cookies` object. You should instead inject an instance of `play.api.mvc.CookieHeaderEncoding` which does not depends on global state.

I will keep this open since we need to add the annotation there.

**Comment by mt40** [comment]
Shouldn't this be closed, it is fixed in #9215
